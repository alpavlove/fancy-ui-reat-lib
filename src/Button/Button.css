.button {
  color: #fff;
  padding: 0.5rem 2rem;
  border: 0;
  border-radius: 0.25rem;
}

.button-primary {
  background-color: #3b3b98;
}

.button-success {
  background-color: #51ff0c;
}
