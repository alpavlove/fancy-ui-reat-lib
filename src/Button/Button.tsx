import React, { ReactNode } from 'react';
import './Button.css';

export type ButtonProps = {
  onClick(): void;
  children: ReactNode;
  variant?: 'primary' | 'success';
  isDisabled?: boolean;
};

export function Button({
  children,
  onClick,
  isDisabled = false,
  variant = 'primary',
}: ButtonProps) {
  const className = `button button-${variant}`;
  return (
    <button onClick={onClick} disabled={isDisabled} className={className}>
      {children}
    </button>
  );
}
