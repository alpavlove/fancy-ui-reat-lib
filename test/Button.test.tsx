import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from '../src/Button/Button';

describe('Button', () => {
  describe('should be rendered correctly', () => {
    test('default', () => {
      render(<Button onClick={console.log}>Click me</Button>);
      const button = screen.getByText('Click me');
      expect(button.classList.contains('button-primary')).toBeTruthy();
    });

    test('success', () => {
      const { container } = render(
        <Button onClick={console.log} variant="success">
          Click me
        </Button>
      );
      const button = container.querySelector('button');
      expect(button!.classList.contains('button-success')).toBeTruthy();
    });

    test('disabled', () => {
      const { asFragment } = render(
        <Button onClick={console.log} isDisabled={true}>
          Click me
        </Button>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  test('should be clickable', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Click me</Button>);
    fireEvent.click(screen.getByText('Click me'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test('should not be clickable if disabled', () => {
    const onClick = jest.fn();
    render(
      <Button onClick={onClick} isDisabled={true}>
        Click me
      </Button>
    );
    fireEvent.click(screen.getByText('Click me'));
    expect(onClick).not.toHaveBeenCalled();
  });
});
