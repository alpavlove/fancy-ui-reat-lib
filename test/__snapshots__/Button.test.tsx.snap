// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be rendered correctly disabled 1`] = `
<DocumentFragment>
  <button
    class="button button-primary"
    disabled=""
  >
    Click me
  </button>
</DocumentFragment>
`;
